Patch offered & accepted upstream for a release after 47.0 in:
https://gitlab.gnome.org/GNOME/gdm/-/merge_requests/267

From 2fd38433e149a75a71c71f68a3a0ceb04055d4a5 Mon Sep 17 00:00:00 2001
From: Alan Coopersmith <alan.coopersmith@oracle.com>
Date: Fri, 13 Sep 2024 16:02:45 -0700
Subject: [PATCH 2/7] gdm-session-solaris-auditor: include "gdm-common.h"
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Fixes build error with gcc 14:
../daemon/gdm-session-solaris-auditor.c:331:17: error: implicit declaration
 of function ‘gdm_get_pwent_for_name’ [-Wimplicit-function-declaration]

Signed-off-by: Alan Coopersmith <alan.coopersmith@oracle.com>
---
 daemon/gdm-session-solaris-auditor.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/daemon/gdm-session-solaris-auditor.c b/daemon/gdm-session-solaris-auditor.c
index 8390ab3fe..9f8ff8f89 100644
--- a/daemon/gdm-session-solaris-auditor.c
+++ b/daemon/gdm-session-solaris-auditor.c
@@ -25,6 +25,7 @@
  */
 #include "config.h"
 #include "gdm-session-solaris-auditor.h"
+#include "gdm-common.h"
 
 #include <syslog.h>
 #include <security/pam_appl.h>
-- 
2.45.2

